app/models/service/containers.rb

```diff
@@ -6,28 +6,41 @@ def errors
 
     def start_container!(wizard_state)
       ActiveRecord::Base.transaction do
-        container = Container.new(wizard_state.container_attributes) do |r|
-          # eagerly load environment variables
-          state = DockerContainerWizardState.includes(:environment => [:environment_variables])
-                  .find(wizard_state.id)
-          state.environment_variables.each do |environment_variable|
-            r.environment_variables.build :name     => environment_variable.name,
-                                          :value    => environment_variable.value,
-                                          :priority => environment_variable.priority
-          end
-        end
-        Taxonomy.enabled_taxonomies.each do |taxonomy|
-          container.send(:"#{taxonomy}=", wizard_state.preliminary.send(:"#{taxonomy}"))
-        end
-
-        fail ActiveRecord::Rollback unless pull_image(container) && start_container(container)
+        container = create_container_object(wizard_state)
+        container.save!
```

Comment:
I'd like ```save!``` to generate an exception (http://apidock.com/rails/ActiveRecord/Persistence/save%21), I want to make sure all validations are passed when saving. Rails will generate the correct error for the user if validations failed.